File: wrappers/kotlin/src/commonMain/kotlin/api/RevocationRegistryDefinitionPrivate.kt
package anoncreds.api

import anoncreds_rs.ObjectHandle
import anoncreds_rs.anoncreds_revocation_registry_definition_private_from_json

class RevocationRegistryDefinitionPrivate
internal constructor(
    handle: ObjectHandle
) : AnoncredsObject(handle) {

    companion object{
        fun fromJson(json: String): RevocationRegistryDefinitionPrivate{
            return fromJson(
                json,
                ::RevocationRegistryDefinitionPrivate,
                ::anoncreds_revocation_registry_definition_private_from_json
            )
        }
    }
}

File: wrappers/dotnet/lib/Models/RevocationRegistryDefinitionPrivate.cs
namespace AnonCredsNet.Models;

public class RevocationRegistryDefinitionPrivate : AnonCredsObject
{
    internal RevocationRegistryDefinitionPrivate(long handle)
        : base(handle) { }

    public static RevocationRegistryDefinitionPrivate FromJson(string json) =>
        FromJson<RevocationRegistryDefinitionPrivate>(json);
}
